From 7999ecd5ee4ea3123f7e75634d2bc57f57ca7070 Mon Sep 17 00:00:00 2001
From: Justin Michaud <jmichaud@igalia.com>
Date: Wed, 6 Aug 2025 21:14:26 +0300
Subject: [PATCH] REGRESSION(2.48.5): [WPE][GTK] Does not compile on ARMv7
 https://bugs.webkit.org/show_bug.cgi?id=296921

Unreviewed build fix.

* Source/JavaScriptCore/llint/WebAssembly.asm: Replace addq with addp
  for sp on armv7

Canonical link: https://commits.webkit.org/290945.344@webkitglib/2.48

Upstream-Status: Backport [https://github.com/WebKit/WebKit/commit/7999ecd5ee4ea3123f7e75634d2bc57f57ca7070]
Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>
---
 Source/JavaScriptCore/llint/WebAssembly.asm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/llint/WebAssembly.asm b/Source/JavaScriptCore/llint/WebAssembly.asm
index 1ac3e2accf3c..bd9041404eb1 100644
--- a/Source/JavaScriptCore/llint/WebAssembly.asm
+++ b/Source/JavaScriptCore/llint/WebAssembly.asm
@@ -736,7 +736,7 @@ if JSVALUE64
     storep memoryBase, Callee[cfr]
 else
     loadp [sp], ws0
-    addq 2 * SlotSize, sp
+    addp 2 * SlotSize, sp
     storep ws0, Callee[cfr]
 end
 
